graph/graph.py

```diff
@@ -23,13 +24,65 @@ class KytosGraph:
 
     def __init__(self):
         self.graph = nx.Graph()
+        self._accepted_metadata = {
+            'ownership',
+            'bandwidth',
+            'reliability',
+            'priority',
+            'utilization',
+            'delay',
+        }
+        ownership_processor = ProcessEdgeAttribute(
+            'ownership',
+            TypeDifferentiatedProcessor({
+                str: lambda x: frozenset(x.split()),
+                dict: lambda x: frozenset(x.keys()),
+                type(None): None
+            })
+        )
         self._filter_functions = {
-            "ownership": lazy_filter(str, filter_in("ownership")),
-            "bandwidth": lazy_filter((int, float), filter_ge("bandwidth")),
-            "reliability": lazy_filter((int, float), filter_ge("reliability")),
-            "priority": lazy_filter((int, float), filter_le("priority")),
-            "utilization": lazy_filter((int, float), filter_le("utilization")),
-            "delay": lazy_filter((int, float), filter_le("delay")),
+            "ownership": EdgeFilter(
+                operator.and_,
+                UseValIfNone(ownership_processor),
+                TypeCheckPreprocessor(
+                    str,
+                    lambda val: frozenset(val.split(','))
+                )
+            ),
+            "not_ownership": EdgeFilter(
+                lambda a, b: not (a & b),
+                UseDefaultIfNone(ownership_processor, frozenset()),
+                TypeCheckPreprocessor(
+                    str,
+                    lambda val: frozenset(val.split(','))
```

Comment:
@Ktmi, when I was reading this code, I found myself multiple times asking if it was checking the expected type for a value that is coming from the API and will be passed down to the `filter`, or if it was type checking and mapping an edge attribute as you've done with `ProcessEdgeAttribute` and `TypeDifferentiatedProcessor` for `ownership`, the latter is clear, but the former isn't, especially folks who are reading this for the first time or might need to patch or augment in the future.

That said, back then we didn't have any validation on API endpoints, so for the values passed down to the `filter` from the mandatory and flexible metrics now we have a stronger guarantee, so I wonder if at this point in time we can get rid of the additional type checking here to make the code even easier to reason. Can you simplify this part, and only keeping maping the str into a set part? We can still have metadata improperly configured, those type errors with the comparator should still be handled though, which form I followed is already in place.

Reply:
With the way I have the code written at this point, it would probably be for the best to just dedicate the TypeCheckPreprocessor to type checking, and not modify the val. I was already planning on converting the value preprocessor on `not_ownership` into a type differentiated processor, in order to also allow for lists of strings.